test/generate_parse_fixture_yml.py

```diff
@@ -1,60 +1,142 @@
 """Utility to generate yml files for all the parsing examples."""
 import multiprocessing
 import os
+import fnmatch
+import re
+import click
+from typing import Callable, Dict, List, NamedTuple, Optional, TypeVar
+
 
 import yaml
 
 from conftest import compute_parse_tree_hash, get_parse_fixtures, parse_example_file
 from sqlfluff.core.errors import SQLParseError
 
 
-def generate_parse_fixture(example):
+S = TypeVar("S")
+
+
+def distribute_work(work_items: List[S], work_fn: Callable[[S], None]) -> None:
+    """Distribute work and ignore results."""
+    with multiprocessing.Pool(multiprocessing.cpu_count()) as pool:
+        for _ in pool.imap_unordered(work_fn, work_items):
+            pass
+
+
+class _ParseExample(NamedTuple):
+    dialect: str
+    sqlfile: str
+
+
+def _create_yaml_path(example: _ParseExample) -> str:
+    dialect, sqlfile = example
+    root = sqlfile[:-4]
```

Comment:
I realize this was pre-existing code that you moved, but could we use `os.path.splitext()` here instead?

Reply:
Done